exporter/azuremonitorexporter/trace_to_envelope.go

```diff
@@ -0,0 +1,660 @@
+		sb.WriteString(netPeerPortAsString)
+		data.Target = sb.String()
+	} else if attrs.HTTPScheme != "" && attrs.NetworkAttributes.NetPeerIP != "" && netPeerPortAsString != "" && attrs.HTTPTarget != "" {
+		sb.WriteString(attrs.HTTPScheme)
+		sb.WriteString("://")
+		sb.WriteString(attrs.NetworkAttributes.NetPeerIP)
+		sb.WriteString(":")
+		sb.WriteString(netPeerPortAsString)
+		sb.WriteString(attrs.HTTPTarget)
+		data.Data = sb.String()
+
+		sb.Reset()
+		sb.WriteString(attrs.NetworkAttributes.NetPeerIP)
+		sb.WriteString(":")
+		sb.WriteString(netPeerPortAsString)
+		data.Target = sb.String()
+	}
+}
+
+// Maps RPC Server Span to AppInsights RequestData
+// https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/rpc.md
+func fillRequestDataRPC(span pdata.Span, data *contracts.RequestData) {
+	attrs := copyAndExtractRPCAttributes(span.Attributes(), data.Properties, data.Measurements)
+
+	var sb strings.Builder
+
+	sb.WriteString(attrs.RPCSystem)
+	sb.WriteString(" ")
+	sb.WriteString(data.Name)
+
+	// Prefix the name with the type of RPC
+	data.Name = sb.String()
+
+	// Set the .Data property to .Name which contain the full RPC method
+	data.Url = data.Name
+
+	// data.Source should be the net.peer.name if available otherwise fallback to net.peer.ip
+	if attrs.NetworkAttributes.NetPeerName != "" {
+		data.Source = attrs.NetworkAttributes.NetPeerName
+	} else if attrs.NetworkAttributes.NetPeerIP != "" {
+		data.Source = attrs.NetworkAttributes.NetPeerIP
+	}
+}
+
+// Maps RPC Client Span to AppInsights RemoteDependencyData
+// https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/rpc.md
+func fillRemoteDependencyDataRPC(span pdata.Span, data *contracts.RemoteDependencyData) {
+	attrs := copyAndExtractRPCAttributes(span.Attributes(), data.Properties, data.Measurements)
+
+	var sb strings.Builder
+
+	// Set the .Data property to .Name which contain the full RPC method
+	data.Data = data.Name
+
+	data.Type = attrs.RPCSystem
+
+	// Favor name over IP for data.Target
+	if attrs.NetworkAttributes.NetPeerName != "" {
+		sb.WriteString(attrs.NetworkAttributes.NetPeerName)
+	} else if attrs.NetworkAttributes.NetPeerIP != "" {
+		sb.WriteString(attrs.NetworkAttributes.NetPeerIP)
+	}
+
+	if attrs.NetworkAttributes.NetPeerPort != 0 {
+		sb.WriteString(":")
+		sb.WriteString(strconv.FormatInt(attrs.NetworkAttributes.NetPeerPort, 10))
+	}
+
+	data.Target = sb.String()
+}
+
+// Maps Database Client Span to AppInsights RemoteDependencyData
+// https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/database.md
+func fillRemoteDependencyDataDatabase(span pdata.Span, data *contracts.RemoteDependencyData) {
+	attrs := copyAndExtractDatabaseAttributes(span.Attributes(), data.Properties, data.Measurements)
+
+	data.Type = attrs.DBType
+	data.Data = attrs.DBStatement
+	data.Target = attrs.DBURL
```

Comment:
https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/database.md#call-level-attributes

* data.Type : DBType seems deprecated. need to switch to `db.system`
* data.Data : we need to look at `db.operation` as a fallback.
* data.Target: `db.url` is deprecated so that we need to use net.peer.*
for example, data.Target will be
1. `net.peer.name:net.peer.port`
2. `net.peer.ip:net.peer.port`
3. `net.peer.name`
4. `net.peer.ip`